src/commands/deploys.rs

```diff
@@ -0,0 +1,154 @@
+                        .help("Optional human readable name for this deployment."),
+                )
+                .arg(
+                    Arg::new("url")
+                        .long("url")
+                        .short('u')
+                        .value_name("URL")
+                        .help("Optional URL that points to the deployment."),
+                )
+                .arg(
+                    Arg::new("started")
+                        .long("started")
+                        .value_name("TIMESTAMP")
+                        .validator(validate_timestamp)
+                        .help("Optional unix timestamp when the deployment started."),
+                )
+                .arg(
+                    Arg::new("finished")
+                        .long("finished")
+                        .value_name("TIMESTAMP")
+                        .validator(validate_timestamp)
+                        .help("Optional unix timestamp when the deployment finished."),
+                )
+                .arg(
+                    Arg::new("time")
+                        .long("time")
+                        .short('t')
+                        .value_name("SECONDS")
+                        .validator(validate_int)
+                        .help(
+                            "Optional deployment duration in seconds.{n}\
+                            This can be specified alternatively to `--started` and `--finished`.",
+                        ),
+                ),
+        )
+        .subcommand(
+            Command::new("list")
+                // Backward compatibility with `releases deploys <VERSION>` commands.
+                .arg(Arg::new("version").long("version").hide(true))
+                .about("List all deployments of a release."),
+        )
+}
+
+fn execute_deploys_new(matches: &ArgMatches) -> Result<()> {
+    let config = Config::current();
+    let api = Api::current();
+    let version = config.get_release_with_legacy_fallback(matches)?;
+    let mut deploy = Deploy {
+        env: matches.value_of("env").unwrap().to_string(),
+        name: matches.value_of("name").map(str::to_owned),
+        url: matches.value_of("url").map(str::to_owned),
+        ..Default::default()
+    };
+
+    if let Some(value) = matches.value_of("time") {
+        let finished = Utc::now();
+        deploy.finished = Some(finished);
+        deploy.started = Some(finished - Duration::seconds(value.parse().unwrap()));
+    } else {
+        if let Some(finished_str) = matches.value_of("finished") {
+            deploy.finished = Some(get_timestamp(finished_str)?);
+        } else {
+            deploy.finished = Some(Utc::now());
+        }
+        if let Some(started_str) = matches.value_of("started") {
+            deploy.started = Some(get_timestamp(started_str)?);
+        }
+    }
+
+    let org = config.get_org(matches)?;
+    let deploy = api.create_deploy(&org, &version, &deploy)?;
```

Comment:
nit: It's confusing to me to redeclare the variable here, can we make the difference more clear?